### PR TITLE
Data collection forms

### PR DESCRIPTION
**Pull request by apanjain**

### CODE REVIEW

**Comment by rohitjha941** [comment]
API for Mentor intern
```{
	"mentor_id" : "1",
	"interns" : [{
		"company" : "Test1q3",
		"duration": "tawe gduawedgyuwAD",
		"domain": "kgvbhh"
	}, 
    {
		"company" : "Test1q3",
		"duration": "tawe gduawedgyuwAD",
        "domain": "kgvbhh"
    }]
}```

File: frontend/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,16583 @@
+{
```
Comment (hacksdump) [inline frontend/package-lock.json:1]: @rohitjha941 We are using yarn package manager. Adding this `package-lock.json` will create conflicts.
Please remove this and make sure the `yarn.lock` works fine.

File: frontend/src/routes/DataCollection/MentorForm.js [hunk 2]
```diff
@@ -0,0 +1,588 @@
...
+      resume,
+      facebook,
+      linkden,
+      groups,
+      achievements,
+      internships,
+    } = this.state;
+
+    const intrestToCreate = interest.filter((i) => typeof i === "string");
+    let createdIntrest = interest.filter((i) => typeof i === "number");
```
Comment (hacksdump) [inline frontend/src/routes/DataCollection/MentorForm.js:152]: fix typo: createdInterest

File: frontend/src/routes/DataCollection/MentorForm.js [hunk 3]
```diff
@@ -0,0 +1,588 @@
...
+
+    // TODO: Make a API to create interests in one GO.
+    intrestToCreate.forEach((interest) => {
+      const data = { interest_name: interest };
+      axios
```
Comment (hacksdump) [inline frontend/src/routes/DataCollection/MentorForm.js:157]: Move API call to `src/api/methods/index.js

File: frontend/src/routes/DataCollection/MentorForm.js [hunk 4]
```diff
@@ -0,0 +1,588 @@
...
+    data.append("interest", createdIntrest);
+    data.append("email", email);
+    data.append("mobile", mobile);
+    data.append("photo", image);
+    data.append("resume", resume);
+    data.append("facebook", facebook);
+    data.append("linkden", linkden);
+    data.append("groups", groups);
+
+    axios
```
Comment (hacksdump) [inline frontend/src/routes/DataCollection/MentorForm.js:180]: Write this API call as a function accepting required parameters in `src/api/methods/index.js` and obtain the response here and act on it accordingly.

File: frontend/src/routes/DataCollection/MentorForm.js [hunk 5]
```diff
@@ -0,0 +1,588 @@
...
+          .then((response) => {
+            console.log(response);
+            this.setState({ errMsg: "Submitted Successfully" });
+          })
+          .catch((error) => {
+            console.log(error);
+            this.setState({ errMsg: "Unable to Submit Right Now" });
+          });
+
+        axios
```
Comment (hacksdump) [inline frontend/src/routes/DataCollection/MentorForm.js:208]: encapsulate all of these axios calls in a single function and import the function in this class.

File: frontend/src/routes/DataCollection/MentorForm.js [hunk 6]
```diff
@@ -0,0 +1,588 @@
...
+                Photograph<span className="color-red">*</span>
+              </label>
+              <input
+                type="file"
+                className="form-control-file"
+                name="photo"
+                aria-describedby="photo-help"
+                id="photo"
+                required
+                onChange={this.handleImage}
```
Comment (hacksdump) [inline frontend/src/routes/DataCollection/MentorForm.js:421]:
```suggestion
                onChange={this.handleImage}
                accept="image/*"
```

File: frontend/package-lock.json (hunk 1 above)
Comment (rohitjha941) [reply frontend/package-lock.json:1]: Done

**Comment by rohitjha941** [comment]
@apanjain Carry out changes in backend. I have a submission.

File: frontend/public/index.html [hunk 7]
```diff
@@ -9,6 +9,13 @@
   <meta name="description" content="Student Mentorship Program" />
   <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
   <meta meta name="viewport" content="width=device-width, user-scalable=no" />
+  
+  <!-- BootStrap -->
+  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css">
```
[the remark this answers is not in the document]
Comment (apanjain) [reply frontend/public/index.html]: Done

File: backend/pages/views.py [hunk 8]
```diff
@@ -20,84 +20,141 @@
...
+        for intern in interns:
+            MentorIntern.objects.create(
+                mentor=mentor,
+                company = intern['company'],
+                duration = intern['duration'],
+                domain = intern['domain']
+            )
+        return Response(status=status.HTTP_201_CREATED)
+
+class InterestView (generics.ListCreateAPIView):
```
[the remark this answers is not in the document]
Comment (apanjain) [reply backend/pages/views.py:98]: Done

File: backend/pages/models.py [hunk 9]
```diff
@@ -67,78 +72,105 @@ def __str__(self):
...
+    def __str__(self):
+        return self.group_name
+
 
 class Interest(models.Model):
     interest_name = models.CharField(
-        max_length = 100,
-        default = ""
+        max_length=100,
+        default=""
```
[the remark this answers is not in the document]
Comment (apanjain) [reply backend/pages/models.py]: done

File: frontend/src/routes/DataCollection/MentorForm.js (hunk 2 above)
Comment (apanjain) [reply frontend/src/routes/DataCollection/MentorForm.js]: done

File: frontend/src/routes/DataCollection/MentorForm.js (hunk 3 above)
Comment (apanjain) [reply frontend/src/routes/DataCollection/MentorForm.js]: done

File: frontend/src/routes/DataCollection/MentorForm.js (hunk 5 above)
Comment (apanjain) [reply frontend/src/routes/DataCollection/MentorForm.js]: done

File: frontend/src/routes/DataCollection/MentorForm.js (hunk 4 above)
Comment (apanjain) [reply frontend/src/routes/DataCollection/MentorForm.js]: done

**Comment by apanjain** [comment]
@rohitjha941 i'm still getting the same error on submission of form. Please look into it

**Comment by apanjain** [comment]
@hacksdump the "response" variable in MentorForm.js line:172 should only update on completion of all the post requests. Make changes accordingly

**Comment by rohitjha941** [comment]
@apanjain Resolve conflict & Run Linter

**Pull request merged**